tests/components/vesync/test_humidifier.py

```diff
@@ -0,0 +1,217 @@
+    SERVICE_SET_MODE,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+
+from tests.common import MockConfigEntry
+
+
+async def test_humidifier_state(
+    hass: HomeAssistant, humidifier_config_entry: MockConfigEntry
+) -> None:
+    """Test the resulting setup state is as expected for the platform."""
+
+    humidifier_id = "humidifier.humidifier_200s"
+    expected_entities = [
+        humidifier_id,
+        "sensor.humidifier_200s_humidity",
+    ]
+
+    assert humidifier_config_entry.state is ConfigEntryState.LOADED
+
+    for entity_id in expected_entities:
+        assert hass.states.get(entity_id).state != STATE_UNAVAILABLE
+
+    assert hass.states.get("sensor.humidifier_200s_humidity").state == "35"
+
+    state = hass.states.get(humidifier_id)
+
+    # ATTR_HUMIDITY represents the target_humidity which comes from configuration.auto_target_humidity node
+    assert state.attributes.get(ATTR_HUMIDITY) == 40
+
+
+@patch(
+    "homeassistant.components.vesync.humidifier.VeSyncHumidifierHA.schedule_update_ha_state"
+)
+async def test_set_target_humidity_invalid(
+    mock_schedule_update_ha_state,
+    hass: HomeAssistant,
+    humidifier_config_entry: MockConfigEntry,
+) -> None:
+    """Test handling of invalid value in set_humidify method."""
+
+    humidifier_entity_id = "humidifier.humidifier_200s"
+
+    # Setting value out of range results in ServiceValidationError and
+    # VeSyncHumid200300S.set_humidity does not get called.
+    with patch("pyvesync.vesyncfan.VeSyncHumid200300S.set_humidity") as method_mock:
+        with pytest.raises(ServiceValidationError):
+            await hass.services.async_call(
+                HUMIDIFIER_DOMAIN,
+                SERVICE_SET_HUMIDITY,
+                {ATTR_ENTITY_ID: humidifier_entity_id, ATTR_HUMIDITY: 20},
+                blocking=True,
+            )
+        await hass.async_block_till_done()
+        method_mock.assert_not_called()
+        mock_schedule_update_ha_state.assert_not_called()
+
+
+@patch(
+    "homeassistant.components.vesync.humidifier.VeSyncHumidifierHA.schedule_update_ha_state"
+)
+@pytest.mark.parametrize(
+    "success",
+    [False, True],
+)
+async def test_set_target_humidity_VeSync(
+    mock_schedule_update_ha_state,
+    hass: HomeAssistant,
+    humidifier_config_entry: MockConfigEntry,
+    success: bool,
+) -> None:
+    """Test handling of return value from VeSyncHumid200300S.set_humidity."""
+
+    humidifier_entity_id = "humidifier.humidifier_200s"
+
+    # If VeSyncHumid200300S.set_humidity fails (returns False), then HomeAssistantError is raised
+    # and schedule_update_ha_state is not called.
+    expectation = nullcontext() if success else pytest.raises(HomeAssistantError)
```

Comment:
Please don't have if statements in the tests as they make it harder to see what is tested

Reply:
I converted the tests to use parameters "api_response", "expectation" which I think makes it all more cleaner.